src/SWP/Bundle/CoreBundle/Controller/WidgetController.php

```diff
@@ -79,12 +78,8 @@ public function listAction(Request $request)
      */
     public function getAction(Request $request, $id)
     {
-        if (!$id) {
-            throw new UnprocessableEntityHttpException('You need to provide widget Id (integer).');
-        }
-
         $entityManager = $this->get('doctrine')->getManager();
-        $widget = $entityManager->getRepository('SWP\Bundle\CoreBundle\Model\WidgetModel')->getById($id)->getOneOrNullResult();
+        $widget = $entityManager->getRepository('SWP\Bundle\CoreBundle\Model\WidgetModel')->getById($id)->getQuery()->getOneOrNullResult();
```

Comment:
but then we will block it for future usage and modifications, i think that from repository we should always (where it makes sense) return `QueryBuilder` and `getQuery()->getOneOrNullResult()` do in providers. Here we fetch entity by ID so simple `findOneBy` would be enough but then we can't extend it for example in revisions context.